src/Factory.php

```diff
@@ -87,11 +88,23 @@ public function __call(string $name, array $arguments): array
     }
 
     /**
-     * @return Proxy<TObject>&TObject
-     * @phpstan-return Proxy<TObject>
+     * @return (Proxy<TObject>&TObject)|TObject
+     * @phpstan-return ($noProxy is true ? TObject: Proxy<TObject>)
+     * @psalm-suppress ImplementedReturnTypeMismatch
      */
-    final public function create(array|callable $attributes = []): Proxy
+    final public function create(
+        array|callable $attributes = [],
+        /**
+         * @deprecated
+         * @internal
+         */
+        bool $noProxy = false
```

Comment:
we could prevent adding this parameter, and create a method:
```php
public function withProxy(): bool
{
    return $this instanceof PersistentObjectProxyFactory::class;
}
```
but this way, we cannot use the phpstan trick `@phpstan-return ($noProxy is true ? TObject: Proxy<TObject>)`
and it breaks the type system in multiple places :shrug: 

So I'd prefer to use this technique which is far from perfect but which seems acceptable for  a BC layer, WDYT?